photon-server/src/main/java/org/photonvision/common/dataflow/networktables/NTDataPublisher.java

```diff
@@ -0,0 +1,168 @@
+import java.util.function.Supplier;
+import org.photonvision.common.dataflow.CVPipelineResultConsumer;
+import org.photonvision.vision.pipeline.result.CVPipelineResult;
+import org.photonvision.vision.pipeline.result.SimplePipelineResult;
+
+public class NTDataPublisher implements CVPipelineResultConsumer {
+
+    private final NetworkTable rootTable = NetworkTablesManager.kRootTable;
+    private NetworkTable subTable;
+    private NetworkTableEntry rawDataEntry;
+
+    private NetworkTableEntry pipelineIndexEntry;
+    private final Consumer<Integer> pipelineIndexConsumer;
+    private NTDataChangeListener pipelineIndexListener;
+    private NetworkTableEntry driverModeEntry;
+    private final Consumer<Boolean> driverModeConsumer;
+    private NTDataChangeListener driverModeListener;
+
+    private NetworkTableEntry latencyMillisEntry;
+    private NetworkTableEntry hasTargetEntry;
+    private NetworkTableEntry targetPitchEntry;
+    private NetworkTableEntry targetYawEntry;
+    private NetworkTableEntry targetAreaEntry;
+    private NetworkTableEntry targetPoseEntry;
+
+    private final Supplier<Integer> pipelineIndexSupplier;
+    private final BooleanSupplier driverModeSupplier;
+
+    private String currentCameraNickname;
+
+    public NTDataPublisher(
+            String cameraNickname,
+            Supplier<Integer> pipelineIndexSupplier,
+            Consumer<Integer> pipelineIndexConsumer,
+            BooleanSupplier driverModeSupplier,
+            Consumer<Boolean> driverModeConsumer) {
+        this.pipelineIndexSupplier = pipelineIndexSupplier;
+        this.pipelineIndexConsumer = pipelineIndexConsumer;
+        this.driverModeSupplier = driverModeSupplier;
+        this.driverModeConsumer = driverModeConsumer;
+
+        currentCameraNickname = cameraNickname;
+        updateCameraNickname(cameraNickname);
+        updateEntries();
+    }
+
+    private void onPipelineIndexChange(EntryNotification entryNotification) {
+        var newIndex = (int) entryNotification.value.getDouble();
+        var originalIndex = pipelineIndexSupplier.get();
+
+        if (newIndex == originalIndex) {
+            // TODO: Log
+            return;
+        }
+
+        pipelineIndexConsumer.accept(newIndex);
+        var setIndex = pipelineIndexSupplier.get();
+        if (newIndex != setIndex) { // set failed
+            pipelineIndexEntry.forceSetNumber(setIndex);
+            // TODO: Log
+        }
+        // TODO: Log
+    }
+
+    private void onDriverModeChange(EntryNotification entryNotification) {
+        var newDriverMode = entryNotification.value.getBoolean();
+        var originalDriverMode = driverModeSupplier.getAsBoolean();
+
+        if (newDriverMode == originalDriverMode) {
+            // TODO: Log
+            return;
+        }
+
+        driverModeConsumer.accept(newDriverMode);
+        // TODO: Log
+    }
+
+    private void updateEntries() {
+        rawDataEntry = subTable.getEntry("rawData");
```

Comment:
The name of the entry should be `rawBytes` (this is how it is in photonlib).

Reply:
Resolved in 71f99d5